protocol/0087-VAMM-automated_market_maker.md

```diff
@@ -0,0 +1,268 @@
+  
+- A vAMM which has been created and is active contributes with it's proposed fee level to the active fee setting mechanism. (<a name="0087-VAMM-015" href="#0087-VAMM-015">0087-VAMM-015</a>)
+- A vAMM's virtual ELS should be equal to the ELS of a regular LP with the same committed volume on the book (i.e. if a vAMM has an average volume on each side of the book across the epoch of 10k USDT, their ELS should be equal to that of a regular LP who has a commitment which requires supplying 10k USDT who joined at the same time as them). (<a name="0087-VAMM-016" href="#0087-VAMM-016">0087-VAMM-016</a>)
+  - A vAMM's virtual ELS should grow at the same rate as a full LP's ELS who joined at the same time. (<a name="0087-VAMM-017" href="#0087-VAMM-017">0087-VAMM-017</a>)
+- A vAMM can vote in market update proposals with the additional weight of their ELS (i.e. not just from governance token holdings). (<a name="0087-VAMM-018" href="#0087-VAMM-018">0087-VAMM-018</a>)
+
+- If a vAMM is cancelled with `Abandon Position` then it is closed immediately. All funds which were in the `general` account of the vAMM are returned to the user who created the vAMM and the remaining position and margin funds are moved to the network to close out as it would a regular defaulted position. (<a name="0087-VAMM-019" href="#0087-VAMM-019">0087-VAMM-019</a>)
+
+- If a vAMM is cancelled and set in `Reduce-Only` mode when it is currently long, then: (<a name="0087-VAMM-020" href="#0087-VAMM-020">0087-VAMM-020</a>)
+  - It creates no further buy orders even if the current price is above the configured lower price.
+  - When one of it's sell orders is executed it still does not produce buy orders, and correctly quotes sell orders from a higher price.
+  - When the position reaches `0` the vAMM is closed and all funds are released to the user after the next mark to market.
+
+
+- If a vAMM is cancelled and set in `Reduce-Only` mode when it is currently short, then: (<a name="0087-VAMM-021" href="#0087-VAMM-021">0087-VAMM-021</a>)
+  - It creates no further sell orders even if the current price is below the configured upper price.
+  - When one of it's buy orders is executed it still does not produce sell orders, and correctly quotes buy orders from a lower price.
+  - When the position reaches `0` the vAMM is closed and all funds are released to the user after the next mark to market.
+
+- If a vAMM is cancelled and set in `Reduce-Only` mode when it currently has no position then all funds are released after the next mark to market. (<a name="0087-VAMM-022" href="#0087-VAMM-022">0087-VAMM-022</a>)
+
+- If a vAMM is cancelled and set into `Reduce-Only` mode, then an amend is sent by the user who created it, the vAMM is amended according to those instructions and is moved out of `Reduce-Only` mode back into normal operation. (<a name="0087-VAMM-023" href="#0087-VAMM-023">0087-VAMM-023</a>)
```

Comment:
A couple of things that have popped into mind that we will probably want AC's for:
- Amending a vAMM such that when it sends in an order to "pull" it into position, check that it doesn't try to trade with itself i.e the matching process excludes the vAMM being amended.
- If someone uses a key to create a vAMM then uses the same key to submits a normal limit order like a normal person, should it be a considered a wash trade if it trades with the vAMM?